openedx/features/badging/tests/test_views.py

```diff
@@ -72,6 +88,66 @@ def test_trophycase_with_some_earned_badges(self, mock_populate_trophycase):
         badging_views.trophycase(request)
         mock_populate_trophycase.assert_called_once_with(request.user, mock.ANY, [user_badge])
 
+    @ddt.data(STATUS_CODE_TRUE, STATUS_CODE_FALSE)
+    @mock.patch('openedx.features.badging.views.render_to_response')
+    @mock.patch('openedx.features.badging.views.add_posts_count_in_badges_list')
+    @mock.patch('openedx.features.badging.views.NodeBBClient')
+    @mock.patch('openedx.features.badging.views.get_badge_progress_request_data')
+    @mock.patch('openedx.features.badging.views.get_discussion_team_ids')
+    def test_trophycase_get_progress_api(self, api_status_code, mock_get_discussion_team_ids,
+                                         mock_get_badge_progress_request_data,
+                                         mock_nodebb_client,
+                                         mock_add_posts_count_in_badges_list,
+                                         mock_render_response):
+        """
+        Test nodeBB api client call arguments and its response. Api will node be called it is mocked so we are just
+        setting response of api and later check if that response is using or not.
+        :param api_status_code: Status code that will be used to test both positive and negative case
+        :param mock_get_discussion_team_ids: mock get_discussion_team_ids helper method
+        :param mock_get_badge_progress_request_data: mock get_badge_progress_request_data helper method
+        :param mock_nodebb_client: mock NodeBB Client class
+        :param mock_add_posts_count_in_badges_list: mock add_posts_count_in_badges_list helper method
+        :param mock_render_response: mock render_to_response method because we are having problems with our templates
+        :return: None
+        """
+        request = self.request_factory.get(reverse('trophycase'), {'json': True})
```

Comment:
We no longer need `{'json': True}`. Can you please remove this from other tests as well